### PR TITLE
Adds SG-62 Target Rifle, a new SG gun with an underbarreled spotting rifle. MK2

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request

This is a reopening of #12580, with new sprites.

image
New gun for SGs- The SG-62 Target Rifle. It has a forced suppressor, underbarreled spotting rifle with several roundtypes. Thanks to Memeager for the sprites and Elsa for commissioning and letting me use them.
I 'updated' the sprites into a more fitting version of our style.

(if you want full detailed stats, look below)
<details>

SG-62 Target Rifle
-
Generally based off of DMR statistics in general, seeking to be a midrange fighter focusing more on quality of fire rather than quantity of fire.

(Left is Target, Right is DMR)

Damage 
40/30/5 (DMG/AP/SUNDER)| DMR-> 65/15/2 (Reference)

Gun handling

Fire Delay-> 5|.65
Aim slowdown-> 0.85|.75
Wield delay-> 0.75|0.8
IFF (Its.. an SG gun, also uses SG skill as an aside.)
Movement accuracy penalties->8
Acc mult->1.25
Scatter = 0
Minimum range-> 3|5
Attachables-> Only rail attachments, the suppressor (which is a prop and not an attachment) and underbarreled spotting gun are irremoveable. Most noticeable of all is it has a rail scope, which may be removed in favor of mini only.

Underbarreled SR-153 spotting rifle
-

An underbarreled spotting rifle that applies specialized rounds that apply status effects on your foes, could do several things in general, it is irremoveable.
It is also IFF capable.

2 seconds of fire delay.

Round types
-
70 damage 20 AP 5 sunder [Normal]
1 stagger + 1 slow [High Impact]
1 weaken + 1 slow [Heavy Rubber]
Plasmaloss (20% + 20) [Tanglefoot]
4 knockback [Tungsten]
Incendary effect [Incend]
60 Damage+Airburst [Flak]
</details>

Changes compared to before below.
<details>

- Target Rifle Mags no longer have a reload delay.
- SR has had its fire delay halved, Target Rifle has had its fire delay decreased from .55 to .5.
- Point buy cost for a mag has been reduced from 5 to 3.
- Req costs for a mag for TR has been reduced from 40 to 35, and Scout rifle mags reduced to 15.
- Shell speeds are 4 instead of 3. This is something I missed as most marksman wepaons are 4 shell speed. Applies to both Target Rifle and Spotting Rifle.

</details>

## Why It's Good For The Game
Adds a mid-long range focused SG gun to supplement with more quality of fire and gimmick underbarrel compared to the area suppression of the Gatling SG and the normal SG, this combined with the spotting rifle should allow for fun combinations and interesting gameplay compared to the normal SG and Gatling SG.

## Changelog
:cl:
add: Added new SG gun, the SG-62 target rifle. It is a rifle with a unbarreled spotting rifle that applies bullets with status effects or just normal bullets that deal a lot of damage, it has an irremoveable suppressor as well.
/:cl:

### CODE REVIEW

**Comment by Derrick2Electricboogaloo** [comment]
So.. Just the T-50 again?

**Comment by Mrrpip** [comment]
I will say, the last attempt's issue with ammo economy might be able to be offset with more ease, thanks to the unlimited ammo (Assuming you have 9mm AP boxes and don't lose the mags) SG pistol being a class point buy.

**Comment by FosterGrime-Heymynameslouie** [comment]
the penis colored gun returns

**Comment by Sebbe9123** [comment]
How much ammo does each magazine have?

**Comment by FIXTHERADBOXPLEASE** [comment]
It's a marksman rifle, not an assault rifle. Also the main gun has 40 round magazines, and the spotting rifle has 5 round magazines

**Review by Kuro020 — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]
Fix conflict and ping me for tmc

**Comment by yyzsong** [comment]
Please give this thing an NVG scope.

**Comment by isotope436** [comment]
> 

just used it you can snipe easily with it any issue is souly skill as this already has HV ammo in it to counter the suppressor. 

also the flak rounds still dont work lmao

**Comment by yyzsong** [comment]
OK
this gun has the weirdest bug ever - it converts the 10x26 rounds that come with the mag into 10x27 rounds.

Reproduction steps:
1. obtain empty target rifle
2. obtain target rifle magazine
3. pull a handful of rounds out and see that they are 10x26
4. put them back in the mag, reinsert magazine into target rifle
5. eject the mag and then eject the chambered round
6. it's 10x27.

**Comment by Lumipharon** [comment]
This is basically an SG specific scout rifle. It shoots slightly slower, but has better AP/worse sunder, and has a (probably) better underbarrel).

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -1241,6 +1241,88 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	hud_state = "smartgun"
+	hud_state_empty = "smartgun_empty"
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
+	damage = 40
+	penetration = 30
+	sundering = 5
+	shell_speed = 4
+	damage_falloff = 0.5
+	accurate_range = 25
+	accurate_range_min = 3
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:1256]: This is essentially pointless. At 2 tiles you're in pb range, so a min range of 3 means ONLY at 3 tiles away you get slightly less accuracy.

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -1241,6 +1241,88 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	accuracy = 15
+
+/datum/ammo/bullet/spottingrifle/highimpact
+	name = "smart high-impact spotting bullet"
+	hud_state = "spotrifle_impact"
+	damage = 10
+	sundering = 0.5
+
+/datum/ammo/bullet/spottingrifle/highimpact/on_hit_mob(mob/M,obj/projectile/P)
+	staggerstun(M, P, stagger = 1, slowdown = 1)
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:1277]: All of these have no max range override. As this is supposed to be a long range gun, I assume you want them to work past 5 tiles?

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -1241,6 +1241,88 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+		X.use_plasma(20 + 0.2 * X.xeno_caste.plasma_max * X.xeno_caste.plasma_regen_limit) // This is draining 20%+20 flat per hit.
+
+/datum/ammo/bullet/spottingrifle/tungsten
+	name = "smart tungsten spotting bullet"
+	hud_state = "spotrifle_tungsten"
+	damage = 10
+	sundering = 0.5
+
+/datum/ammo/bullet/spottingrifle/tungsten/on_hit_mob(mob/M,obj/projectile/P)
+	staggerstun(M, P, knockback = 4)
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:1306]: guh, this is meme enough on the tl-127

File: code/modules/projectiles/guns/rifles.dm [hunk 4]
```diff
@@ -1248,6 +1248,83 @@
...
+	attachable_offset = list("muzzle_x" = 12, "muzzle_y" = 22, "rail_x" = 15, "rail_y" = 22, "under_x" = 28, "under_y" = 16, "stock_x" = 12, "stock_y" = 14)
+	starting_attachment_types = list(/obj/item/weapon/gun/rifle/standard_spottingrifle, /obj/item/attachable/stock/strstock)
+
+	fire_delay = 0.5 SECONDS
+	burst_amount = 0
+	accuracy_mult_unwielded = 0.4
+	accuracy_mult = 1.25
+	scatter = -5
+	scatter_unwielded = 20
+	movement_acc_penalty_mult = 8
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/rifles.dm:1295]: The negative base scatter (and extraordinarily high acc bonus) means the movement penalty is entirely cancelled out.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_datums.dm (hunk 1 above)
Comment (Pariah919) [reply code/modules/projectiles/ammo_datums.dm:1256]: I could probably just remove it with that in consideration, honestly.

File: code/modules/projectiles/ammo_datums.dm (hunk 2 above)
Comment (Pariah919) [reply code/modules/projectiles/ammo_datums.dm:1277]: I might slap it with a max range of 9 to 14 tiles. I want it to work past 5 but can't really decide what the max range should be beyond that. I might do 9 or 10.

**Comment by LoganKele** [comment]
I don't know what role this gun takes, but I assume it's a DMR considering it functions similarly to the DMR-37 in the optimal use.

Only issues is that the projectile is short range and on the T-47 given it drops off 6+ tiles below minimum range.

File: code/modules/projectiles/ammo_datums.dm (hunk 2 above)
Comment (LoganKele) [reply code/modules/projectiles/ammo_datums.dm:1277]: > I might slap it with a max range of 9 to 14 tiles. I want it to work past 5 but can't really decide what the max range should be beyond that. I might do 9 or 10.

Wouldn't that make it's role as a DMR pointless? 10 tiles is like 2 tiles below your screen reach.

**Comment by Neskah** [comment]
This might be a long comment. I think this gun is an interesting idea, but needs some tweaking in several spots. I'm not going to discuss any of that, though. Every single caliber related to this gun is so broken that I don't even know what they're supposed to properly be, and it was this way the first time the weapon was test-merged. I'm going to list everything I've found.

- Weapon description states rifle is in "high velocity 10x27mm" and the underslung is in "12x66mm".
- SG-62 magazine states it is chambered in "10x27mm HV", matching description.
- Removing bullets from the SG-62 magazine produces "handful of smart marksman bullet (10x26mm caseless)"
- Racking the SG-62's charging handle ejects a 10x27mm round, which will not fit in the 10x27mm magazine, because the ammunition inside is actually 10x26mm.
- Underslung rifle takes 12x7mm magazines.
- Underslung/SR-153 magazines actually contain 12.7 magnum rounds. (This is true for ALL SR-153 magazine/round variants - the magazine is 12x7mm and the cartridges are 12.7 magnum.)

This is currently every weird caliber-confusion I'm aware of. As far as I can tell, all of the above-mentioned rounds ARE unique to the SG-62. We have several other weapons chambered in 10x26mm and 10x27mm (non-smart), and the R-76 is chambered in 12.7 maxnum, but none of their rounds fit in SG-62/SR-153 mags, and vice versa. (Which is good.)

This is not ultimately game-breaking, as the only real issue is having ejected SG-62 rounds be incompatible with the SG-62 magazines, but it is admittedly extremely confusing and it makes my brain hurt.

File: code/modules/projectiles/ammo_datums.dm (hunk 3 above)
Comment (isotope436) [reply code/modules/projectiles/ammo_datums.dm:1306]: isnt the point of the tungsten rounds to be knockback iv used them ingame they are alright prefer the incin tho 

> guh, this is meme enough on the tl-127

**Comment by isotope436** [comment]
i gave him new sproots now we just need to bully him to actually remember to put them into the pr

**Comment by imsxz** [comment]
image
primordial empress.
fire underbarrel, fire main gun twice, repeat.
150 dps, 15 sunder per second.

i ditched the scope for mag harn+db on back and became an unreasonably powerful frontliner, like the br-8 scout but way more damage

**Comment by Arctrooper8** [comment]
I tested this today and it seems really really strong, I managed to 1v1 a queen and win killing her much to the Queen's surprise in dchat.

The main rifle is decent enough on it's own but more serves as useful filler in-between under barrel shots. 

The under barrel is where things get crazy, depending on what ammo you picked you will either be a monster or a wet noodle.

The damage magazine is huge burst damage which shreds even tougher xenos like queen, and since it's only one fast bullet it's super easy to get full value and scare the shit out of xenos if not outright delete them.

The stagger magazine causes stagger and all the power invested in that, stun magazine is similar causes stun and is powerful because of that.

The knockback magazine is hilarious and has huge amounts of knockback and the stun of being forcefully moved that comes with knockback.

Tanglefoot is strong but it's nothing new and not at all crazy sacrificing damage to reduce a Xeno's capability to use abilities

Airburst is just a 3x3 AoE area of damage and is basically almost useless outside of super niche scenarios such as killing crit xenos or killing crit marines.

Incendiary is similar, being outclassed by the better magazines and just not being good in general the fire being weak and thus rather ignorable by xenos.

Overall this feels like way too much damage, especially in a burst sense. I'm kind of worried it's going to end up like the SG-25 where it's the only gun people pick and they use it to nuke xenos and rush them down.

**Review by Lumipharon — CHANGES_REQUESTED** [review]
I don't really get the base ammo.
The rifle has super high AP and sunder, to the point that the special damage bullet is almost exactly the same. Its not even that its too strong, its just ???

**Review by Kuro020 — CHANGES_REQUESTED** [review]
mob sprites need work

**Comment by Lumipharon** [comment]
MJP gaslit himself.

**Comment by Lumipharon** [comment]
I still think this is a confusing gun for the sg role but eh.

**Pull request merged**